thread.go

```diff
@@ -26,3 +31,69 @@ func (t *Thread) GetSubject() string {
 func (t *Thread) GetID() string {
 	return C.GoString(C.notmuch_thread_get_thread_id(t.cptr))
 }
+
+// Count returns the total number of messages in the current thread.
+func (t *Thread) Count() int {
+	return int(C.notmuch_thread_get_total_messages(t.cptr))
+}
+
+// CountMatched returns the total number of messages in the current thread.
+func (t *Thread) CountMatched() int {
+	return int(C.notmuch_thread_get_matched_messages(t.cptr))
+}
+
+// TopLevelMessages returns an iterator for the top-level messages in the
+// current thread in oldest-first order.
+func (t *Thread) TopLevelMessages() *Messages {
+	return &Messages{
+		thread: t,
+		cptr:   C.notmuch_thread_get_toplevel_messages(t.cptr),
+	}
+}
+
+// Messages returns an iterator for all messages in the current thread in
+// oldest-first order.
+func (t *Thread) Messages() *Messages {
+	return &Messages{
+		thread: t,
+		cptr:   C.notmuch_thread_get_messages(t.cptr),
+	}
+}
+
+// Authors returns the list of authors.
+func (t *Thread) Authors() []string {
```

Comment:
Per the docs in notmuch.h, there's a "|" separating matched vs.
non-matched authors. Wee need to deal with that somehow or we'll get
an author like "Bob|Alice" sometimes.

There's also the qeustion as to whether we want to propagate that information
back to the user. We could return ([]string, []string) maybe?